modules/seqexec/server/src/main/scala/seqexec/server/nifs/NifsEpics.scala

```diff
@@ -111,78 +115,103 @@ class NifsEpics(epicsService: CaService, tops: Map[String, String]) {
+      .map(_.toDouble).value
 
-  def numberOfResets: Option[Int] = Option(dcStatus.getIntegerAttribute("numberOfResets").value)
-    .map(_.toInt)
+  def numberOfResets: F[Option[Int]] =
+    Nested(safeAttribute(dcStatus.getIntegerAttribute("numberOfResets")))
+      .map(_.toInt).value
 
-  def readMode: Option[String] = Option(dcStatus.getStringAttribute("readMode").value)
+  def readMode: F[Option[String]] =
+    safeAttribute(dcStatus.getStringAttribute("readMode"))
 
-  def numberOfFowlerSamples: Option[Int] =
-    Option(dcStatus.getIntegerAttribute("numberOfFowSamples").value).map(_.toInt)
+  def numberOfFowSamples: F[Option[Int]] =
+    Nested(safeAttribute(dcStatus.getIntegerAttribute("numberOfFowSamples")))
+      .map(_.toInt).value
 
-  def coadds: Option[Int] = Option(dcStatus.getIntegerAttribute("coadds").value).map(_.toInt)
+  def coadds: F[Option[Int]] =
+    Nested(safeAttribute(dcStatus.getIntegerAttribute("coadds")))
+      .map(_.toInt).value
 
-  def period: Option[Double] = Option(dcStatus.getDoubleAttribute("period").value).map(_.toDouble)
+  def period: F[Option[Double]] =
+    Nested(safeAttribute(dcStatus.getDoubleAttribute("period")))
+      .map(_.toDouble).value
 
-  def countDown: Option[Double] = Option(dcStatus.getDoubleAttribute("countdown").value)
-    .map(_.toDouble)
+  def countDown: F[Option[Double]] =
+    Nested(safeAttribute(dcStatus.getDoubleAttribute("countdown")))
+      .map(_.toDouble).value
 
-  def numberOfPeriods: Option[Int] = Option(dcStatus.getIntegerAttribute("numberOfPeriods").value)
-    .map(_.toInt)
+  def numberOfPeriods: F[Option[Int]] =
+    Nested(safeAttribute(dcStatus.getIntegerAttribute("numberOfPeriods")))
+      .map(_.toInt).value
 
-  def timeMode: Option[String] = Option(dcStatus.getStringAttribute("timeMode").value)
+  def timeMode: F[Option[String]] =
+    safeAttribute(dcStatus.getStringAttribute("timeMode"))
 
   val dhsConnectedAttr: CaAttribute[DhsConnected] = dcStatus.addEnum[DhsConnected]("dhsConnected",
     s"${NifsTop}sad:dc:dhsConnO", classOf[DhsConnected])
 
-  def dcName: Option[String] = Option(dcStatus.getStringAttribute("name").value)
+  def dcName: F[Option[String]] =
+    safeAttribute(dcStatus.getStringAttribute("name"))
 
-  def exposureMode: Option[String] = Option(dcStatus.getStringAttribute("expMode").value)
+  def exposureMode: F[Option[String]] =
+    safeAttribute(dcStatus.getStringAttribute("expMode"))
 
-  def readTime: Option[Double] = Option(dcStatus.getDoubleAttribute("readTime").value)
-    .map(_.toDouble)
+  def readTime: F[Option[Double]] =
+    Nested(safeAttribute(dcStatus.getDoubleAttribute("readTime")))
+      .map(_.toDouble).value
 
-  def biasPwr: Option[Double] = Option(dcStatus.getDoubleAttribute("biasPwr").value)
-    .map(_.toDouble)
+  def biasPwr: F[Option[Double]] =
+    Nested(safeAttribute(dcStatus.getDoubleAttribute("biasPwr")))
+      .map(_.toDouble).value
```

Comment:
It seems that, for a type `T`,
```
Nested(safeAttribute(dcStatus.getTAttribute("name"))).map(_.toT).value
```
is repeated so many times. Is there a way to reduce all the code copying?

Reply:
Right, we could add some utility methods. I expect to use them in other `XXXEpics` objcets